File: javascript/rna-transcription/rna-transcription.js
var DnaTranscriber = function() {};

DnaTranscriber.prototype.toRna = function(strand) {
  var dnaToRna = {"C":"G","G":"C","A":"U","T":"A"};
  var transcribed = "";
  for (var i=0;i<strand.length;i++) {
    transcribed+=dnaToRna[strand.charAt(i)];
  }
  return transcribed;
};

DnaTranscriber.prototype.toDna = function(strand) {
  var rnaToDna = {"C":"G","G":"C","U":"A","A":"T"};
  var transcribed = "";
  for (var i=0;i<strand.length;i++) {
    transcribed+=rnaToDna[strand.charAt(i)];
  }
  return transcribed;
};

module.exports = DnaTranscriber;

File: python/rna-transcription/rna_transcription.py
def to_rna(dna):
	rna = ""
	for char in dna:
		if char not in ["C","G","T","A"]:
			return ''
		else:
			if char == "C":
				rna += "G"
			if char == "G":
				rna += "C"
			if char == "T":
				rna += "A"
			if char == "A":
				rna += "U"
	return rna